term.tex

```diff
@@ -23,6 +23,7 @@
        | term bin-op term ;
        | term "[" term "]" ; array access
        | "{" term "\with" "[" term "]" "=" term "}" ; array functional modifier
+       | "{" term "\with" "[" range "]" "=" term "}" ; array functional modifier
```

Comment:
Is this really something we want to introduce? I imagine that `{ a \with [ 0..3 ]  = 0}` means "an array identical to `a` except that the first 4 cells are equal to `0`".  Will it be really that useful.

In any case, if we're going to keep it, it should go into `loc.tex`.

Reply:
As far I can remember, array slice updates are implemented into frama-c kernel.  That should appear into the rule.

Reply:
Sorry, I forgot the NOT in the previous sentence: array slice updates are not implemented into frama-c kernel and that should appear into the rule.

Reply:
Note: the implementation of  that rule should take into account the way Frama-C handles the offset extension. The current version of Frama-C transforms  `{ a \with [ idx ].c = 0 }` into  `{ a \with [ idx ] = { a.[ idx ] \with .c = 0 }}`. That transformation is invalid when `idx` is a range: that gives a typing error into the nested update of the field `c`.

Reply:
Sorry for the noice,  array slice updates are implementated and the Frama-C transformation about the offset extension rejects correctly the previous exemple. 

Good, every thing is ok with that.
I noticed also that `{ a \with c.[ range ] = 0 }`  is accepted and transformed into `{ a \with c = { a.c \with [range] = 0 }}`. That is correct.